src/utils/manager.py

```diff
@@ -104,159 +157,47 @@ def supported(self) -> bool:
             else:
                 return True
         except subprocess.CalledProcessError:
-            _logger.warning("Could not detect execution in virtualized environment")
+            _logger.warning("Could not detect execution in virtualized environment.")
             return True
 
-    def fetch(self, target: str) -> Optional[MountInfo]:
-        """Fetch information about a mount.
-
-        Args:
-            target: share mountpoint information to fetch.
-
-        Returns:
-            Optional[MountInfo]: Mount information. None if share is not mounted.
-        """
-        # We need to trigger an automount for the mounts that are of type `autofs`,
-        # since those could contain an unlisted mount.
-        _trigger_autofs()
-
-        for mount in _mounts():
-            if mount.mountpoint == target:
-                return mount
-
-        return None
-
-    def mounts(self) -> List[MountInfo]:
-        """Get all mounts on a machine.
-
-        Returns:
-            List[MountInfo]: All current mounts on machine.
-        """
-        _trigger_autofs()
-
-        return list(_mounts("autofs"))
-
-    def mounted(self, target: str) -> bool:
-        """Determine if mountpoint is mounted.
+    @contextlib.contextmanager
+    def mounts(self, force_mount=False) -> Generator[Mounts, None, None]:
```

Comment:
I prefer `Generator` because it betters describes the intent of a context manager. `Iterable` slightly implies that the object could return multiple `Mounts`, which is not true.